permission_handler_web/lib/permission_handler_web.dart

```diff
@@ -0,0 +1,50 @@
+import 'dart:html' as html;
+import 'dart:async';
+
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:permission_handler_platform_interface/permission_handler_platform_interface.dart';
+import 'package:permission_handler_web/web_delegate.dart';
+
+class WebPermissionHandler extends PermissionHandlerPlatform {
+  WebDelegate? webDelegate;
+
+  static final html.MediaDevices devices = html.window.navigator.mediaDevices!;
+  static final html.Permissions? htmlPermissions =
+      html.window.navigator.permissions;
+
+  static void registerWith(Registrar registrar) {
+    PermissionHandlerPlatform.instance = WebPermissionHandler();
+  }
+
+  @override
+  Future<Map<Permission, PermissionStatus>> requestPermissions(
+      List<Permission> permissions) async {
+    webDelegate ??= WebDelegate(devices, htmlPermissions);
```

Comment:
I preferred this as a way to implement lazy initialization. Considering that Flutter web has some difficulties with initial load times, I though the tradeoff with some code duplication was reasonable.

Here is `firebase_analytics_web` and `firebase_auth_web` doing a similar lazy delegate init:
https://github.com/firebase/flutterfire/blob/4e9bb694ad13cd06ecc6c1936cab0a6b9c77273c/packages/firebase_analytics/firebase_analytics_web/lib/firebase_analytics_web.dart#L22

https://github.com/firebase/flutterfire/blob/4e9bb694ad13cd06ecc6c1936cab0a6b9c77273c/packages/firebase_auth/firebase_auth_web/lib/firebase_auth_web.dart#L151

Reply:
In the Firebase examples you referenced the initialisation logic is more than just a simple instantiation of a type and therefore has a big impact on the initial load time, which is why the lazy initialisation makes perfect sense. On top of that they created a property which takes care of the initialisation which prevents the code duplication.

In this case I am not against the lazy initialisation (in our case I don't think it will matter performance wise), I just wanted to prevent the code duplication.

Reply:
Sounds good then. Thanks for the correction.